rep-0149.rst

```diff
@@ -0,0 +1,973 @@
+contain an arbitrary text providing user more information about the
+deprecation::
+
+  <export>
+    <deprecated>
+      This package will be removed in ROS Hydro. Instead, use package
+      FOO, which provides similar features with a different API.
+    </deprecated>
+  </export>
+
+<message_generator>
+'''''''''''''''''''
+
+The content defines the *identifier* for the language bindings
+generated by this package, i.e. in ``gencpp`` this is set to ``cpp``::
+
+  <export>
+    <message_generator>cpp</message_generator>
+  </export>
+
+<metapackage/>
+''''''''''''''
+
+This empty tag declares a special kind of catkin package used for
+grouping other packages.  Metapackages only provide execution-time
+dependencies.  They cannot be used for catkin builds and compile
+nothing themselves.  Metapackages may not install any code or other
+files, although ``package.xml`` does get installed automatically.
+They can depend on other metapackages, but regular catkin packages
+cannot.
+
+A good use for metapackages is to group the major components of your
+robot and then provide a comprehensive grouping for your whole system.
+Package installation tools like ``apt-get`` or ``yum`` can
+automatically install all the packages on which a metapackage directly
+or indirectly depends.  Metapackages can also be used to resolve
+dependencies declared by legacy rosbuild stacks not yet converted to
+catkin.
+
+Every metapackage must have a ``CMakeLists.txt`` containing these
+commands::
+
+  cmake_minimum_required(VERSION 2.8.3)
+  project(PACKAGE_NAME)
+  find_package(catkin REQUIRED)
+  catkin_metapackage()
+
+Because the metapackage ``CMakeLists.txt`` contains a catkin macro,
+its ``package.xml`` must declare a buildtool dependency on catkin::
+
+  <buildtool_depend>catkin</buildtool_depend>
+
+Additional buildtool, build or test dependencies are not permitted.
+
+Because metapackages only supply execution-time dependencies, they use
+``<exec_depend>`` to list the packages in their group::
+
+  <exec_depend>your_custom_msgs</exec_depend>
+  <exec_depend>your_server_node</exec_depend>
+  <exec_depend>your_utils</exec_depend>
+  <exec_depend>another_metapackage</exec_depend>
+
+
+Compatibility
+=============
+
+.. raw:: html
+
+  <font color="red">
+
+Support for format three
+------------------------
+
+Existing tools supporting up to format two will need to be updated to
+honor the new information provided by format three.
+If they are not checking that the format is two or lower they will
+simply ignore the format three specific information when processing a
+package with a format three manifest.
+
+catkin_prepare_release
+''''''''''''''''''''''
+
+As long as a manifest doesn't specify a `compatibility` version the tools
+doesn't need to change.
+For packages which do specify a `compatibility` version the tool should
+probably by default remove the attribute and only after confirmation from
+the user offer to keep it.
```

Comment:
Just because the previous release of a package was compatible was earlier versions doesn't imply the next release will be too. I think it would be safer to assume that it is not until the maintainer explicitly says otherwise.

Reply:
I would rather see the compatibility version updated each time (same behavior as removing it), rather than removed. So long as the tool is automating it, either way should be fine I just think it would produce a more normal history to continuously update the compatibility version rather than periodically removing it and restoring it. I think it's a reasonable assumption that the default behavior after releasing once with a compatibility version will be to continue using the version even if every release is not compatible.

This seems, however, to be a lot of detail on one tools behavior for this document. It might just be sufficient to say "the tool should default to updating the compatibility version each release, requiring the user to explicitly opt into a backwards-compatible release".